Assets/MRTK/SDK/Experimental/Features/UX/BoundsControl/Visuals/TranslationHandles.cs

```diff
@@ -0,0 +1,338 @@
+                {
+                    faceCenters[i] = VisualUtils.GetFaceCenterPosition(i, ref boundsCorners);
+                }
+            }
+
+            UpdateHandles();
+        }
+
+
+        internal void InitEdgeAxis()
+        {
+            faceAxes = new CardinalAxisType[NumFaces];
+            faceAxes[0] = CardinalAxisType.X;
+            faceAxes[1] = CardinalAxisType.X;
+            faceAxes[2] = CardinalAxisType.Z;
+            faceAxes[3] = CardinalAxisType.Z;
+            faceAxes[4] = CardinalAxisType.Y;
+            faceAxes[5] = CardinalAxisType.Y;
+        }
+
+        internal virtual void Reset(bool areHandlesActive, FlattenModeType flattenAxis)
+        {
+            IsActive = areHandlesActive;
+            cachedFlattenAxis = flattenAxis;
+            if (IsActive)
+            {
+                ResetHandles();
+                int[] flattenedHandles = VisualUtils.GetFlattenedIndices(flattenAxis);
+                if (flattenedHandles != null)
+                {
+                    for (int i = 0; i < flattenedHandles.Length; ++i)
+                    {
+                        handles[flattenedHandles[i]].gameObject.SetActive(false);
+                    }
+                }
+            }
+        }
+
+        internal void Create(ref Vector3[] boundsCorners, Transform parent)
+        {
+            faceCenters = new Vector3[6];
+            CalculateFaceCenters(ref boundsCorners);
+            InitEdgeAxis();
+            CreateHandles(parent);
+        }
+
+        protected virtual void CreateHandles(Transform parent)
+        {
+            for (int i = 0; i < faceCenters.Length; ++i)
+            {
+                GameObject faceCenter = new GameObject();
+                faceCenter.name = "faceCenter_" + i.ToString();
+                faceCenter.transform.position = faceCenters[i];
+                faceCenter.transform.parent = parent;
+
+                Bounds faceCenterBounds = CreateVisual(i, faceCenter);
+                float maxDim = VisualUtils.GetMaxComponent(faceCenterBounds.size);
+                float invScale = maxDim == 0.0f ? 0.0f : config.HandleSize / maxDim;
+                VisualUtils.AddComponentsToAffordance(faceCenter, new Bounds(faceCenterBounds.center * invScale, faceCenterBounds.size * invScale),
+                    config.TranslationHandlePrefabColliderType, CursorContextInfo.CursorAction.Move, config.ColliderPadding, parent, config.DrawTetherWhenManipulating);
+
+                handles.Add(faceCenter.transform);
+            }
+
+            VisualUtils.HandleIgnoreCollider(config.HandlesIgnoreCollider, handles);
+
+            objectsChangedEvent.Invoke(this);
+        }
+
+        protected override void RecreateVisuals()
+        {
+            Debug.Log("RecreateVisuals");
+            for (int i = 0; i < handles.Count; ++i)
+            {
+                // get parent of visual
+                Transform obsoleteChild = handles[i].Find("visuals");
```

Comment:
Same comment here on using a const string instead of hardcoded string

Reply:
Same comment about deferring to @thalbern above

Reply:
replaced by static string in base handles